ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/OnDisk.hs

```diff
@@ -374,16 +370,18 @@ runMock cmd initMock =
               tail
             . dropWhile ((/= r) . blockRealPoint . fst)
 
-        -- The snapshot that the real implementation will write to disk
+        -- The snapshot that the real implementation will possibly write to
+        -- disk. If a snapshot of that slot already exists or it corresponds to
+        -- genesis, no snapshot will be taken.
         --
-        -- The real implementation keeps the length of the snapshots at @k@
-        -- (provided that there are enough blocks). The function
-        -- 'ledgerDbCountToPrune' computes how many (old) blocks should be
-        -- dropped to get back into that range; the last (most recent) block to
-        -- be dropped will become the new anchor. It is the anchor that gets
-        -- written to disk.
-        snapped :: Point TestBlock
-        snapped
+        -- The real implementation keeps @k@ snapshots in memory (provided that
+        -- there are enough blocks). The function 'ledgerDbCountToPrune'
+        -- computes how many (old) blocks/snapshots should be dropped to get
+        -- back into that range; the last (most recent) block to be dropped will
+        -- become the new anchor. It is the anchor that gets written to disk,
+        -- i.e., the ledger snapshot @k@ blocks back from the tip.
```

Comment:
Say that `ledgerDbCountToPrune` tells you how many snapshots too many we have, excluding the anchor. Scratch that, just rewrite it without that function.